pyuvdata/uvflag.py

```diff
@@ -5,58 +5,238 @@
 from __future__ import print_function, division, absolute_import
 import numpy as np
 import os
+import warnings
+import copy
+import six
+from six.moves import map
+
+from .uvbase import UVBase
+from . import parameter as uvp
 from . import UVData
 from . import UVCal
 from . import utils as uvutils
-from . import version as uvversion
 from . import telescopes as uvtel
-import warnings
-import copy
-from six.moves import map
 
 
-class UVFlag(object):
-    """ Object to handle flag arrays and waterfalls. Supports reading/writing,
-    and stores all relevant information to combine flags and apply to data.
+class UVFlag(UVBase):
+    """Object to handle flag arrays and waterfalls for interferometric datasets.
+
+    Supports reading/writing, and stores all relevant information to combine flags and apply to data.
+    Initialization of the UVFlag object requires some parameters. Metadata is copied from input object. If input
+    is subclass of UVData or UVCal, the weights_array will be set to all ones.
+    Input lists or tuples are iterated through, treating each entry with an
+    individual UVFlag init.
+
+    Parameters
+    ----------
+    input : UVData, UVCal, str, list of compatible combination of options
+        Input to initialize UVFlag object. If str, assumed to be path to previously
+        saved UVFlag object. UVData and UVCal objects cannot be directly combined,
+        unless waterfall is True.
+    mode : {"metric", "flag"}, optional
+        The mode determines whether the object has a floating point metric_array
+        or a boolean flag_array. Default is "metric".
+    copy_flags : bool, optional
+        Whether to copy flags from input to new UVFlag object. Default is False.
+    waterfall : bool, optional
+        Whether to immediately initialize as a waterfall object, with flag/metric
+        axes: time, frequency, polarization. Default is False.
+    history : str, optional
+        History string to attach to object. Default is empty string.
+    label: str, optional
+        String used for labeling the object (e.g. 'FM'). Default is empty string.
```

Comment:
If the default is in the definition of the method and if there isn't further explanation of it needed, you don't need to call in out in the doctoring -- the docs will make it clear based on the method definition.